### ISSUE TITLE
[innerText] <source> inside <audio> should always return textContent because it is never being rendered

### ISSUE THREAD

**Issue by timdream** [web-platform-tests/wpt#12580]
I am looking at `getter-tests.js` at [here](https://github.com/web-platform-tests/wpt/blob/87329a1500b1aea8f9cf28b40afb08491e25a5c7/html/dom/elements/the-innertext-idl-attribute/getter-tests.js#L135). According to the DOM spec [here](https://html.spec.whatwg.org/multipage/dom.html#the-innertext-idl-attribute) it asks `innerText` to return the same value as the textContent IDL attribute on this element when the element is not "being rendered".

The tests here (and two lines below for `<video>`) however expect an empty string.

This only passes on Gecko and marked as [failing on Chromium](https://github.com/chromium/chromium/blob/master/third_party/WebKit/LayoutTests/external/wpt/html/dom/elements/the-innertext-idl-attribute/getter-expected.txt#L83).

Should this test be corrected?

**Comment by timdream** [comment]
(Context: I was trying to enable https://bugzilla.mozilla.org/show_bug.cgi?id=1484048 which changes the behavior of Gecko. It would happen to align with Chromium and fail on these two tests.)

**Comment by emilio** [comment]
I filed https://github.com/whatwg/html/issues/3947 for display: contents, but I think this is just a Gecko bug in the innerText implementation. The spec uses `being rendered`, but gecko uses "is or has any ancestor with display: none", which is not the same for fallback content for `<canvas>`, for example.

I think all the tests here should output the same:

```html
<!doctype html>
<video>
  <source style="display: block">
</video>
<audio>
  <source id="audio-source" style="display: block">
</audio>
<canvas>
  <div></div>
</canvas>
<div id="display-none" style="display: none"></div>
<pre>
<script>
  function test(selector) {
    let el = document.querySelector(selector);
    el.textContent = "abc";
    document.writeln("\"" + selector + "\" innerText: \"" + el.innerText + "\", textContent: \"" + el.textContent + "\"");
  }

  test("div");
  test("source");
  test("#audio-source");
  test("#display-none");
</script>
</pre>
```

@annevk do you agree?

**Comment by emilio** [comment]
TL;DR: I think that the test and Gecko should be fixed.

**Comment by emilio** [comment]
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1484855 for that.

**Comment by emilio** [comment]
This is fixed there.

**Comment by annevk** [comment]
Thanks @emilio, sorry for missing your question!
